Issue #16, simplify http call

LinkaiQi opened the pull request:
refer to #16

LinkaiQi committed:
Issue #16, simplify http call

yuenj commented on src/fetch_data.js:
```diff
@@ -28,41 +27,6 @@ export const getChangesetCoverageSummary = changeset =>
\ No newline at end of file
+// get coverage from ActiveData for a particular source file
+export const getFileRevisionCoverage = (revision, path) =>
+  fetch(`${activeData}/query`, { jsonHeaders, method:"POST", body: JSON.stringify(Query.testCoverage(revision, path)) });
```
Nice fix! I tried this but it didn't work for me for some reason.

klahnakoski replied:
Yes, this is a nice line!

LinkaiQi commented on src/components/fileviewer.js:
```diff
@@ -33,29 +33,37 @@ export default class FileViewerContainer extends Component {
-          path={path} 
+        <FileViewerMeta
+          revision={revision}
+          path={path}
```
Sorry for the extra modified lines, I think my editor automatically remove the trailing space occur in JSX

yuenj replied:
My bad, I should have ran yarn lint on the earlier pull request.  I am using Visual Studio editor and thought that it would automatically do the same for me.

LinkaiQi replied:
Oh sorry, I totally forget. I didn't run yarn lint either.

klahnakoski replied:
use `?w=1` at the end of a github url to ignore the space changes.

https://github.com/blog/967-github-secrets

klahnakoski commented on src/components/fileviewer.js:
```diff
@@ -33,29 +33,37 @@ export default class FileViewerContainer extends Component {
-    )
+    FetchAPI.getFileRevisionCoverage(revision, path)
+      .then(response => {
+        if (response.status !== 200) {
```
Put this error handling in the `getFileRevisionCoverage` function.

LinkaiQi replied:
I see what Arman doing is to keep the file_data.js simple and clean, and put the error handling, catch codes in the viewer function. How about create a separate js file to holds the API fetching calls for ucosp project?

klahnakoski replied:
My philosophy is to make a facade for every library function you use.  This will isolate you from inevitable anomalies those libraries have in the corner cases.  This facade can be further enhanced it to deal with your application-specific usage patterns.

You will now see that `fetch_data.getRawFile` and `fetch_data.getFileRevisionCoverage` have common code.  Put this common code in its own function (maybe called `fetch_data..post`), which will be the beginning of the "facade" I talk about.  

When you have done that, the `fetch_data.getRawFile` and `fetch_data.getFileRevisionCoverage` should be single-line functions that are used in only one place in your code:  Inline them.

(edit: changed "FetchAPI" to "fetch_data")

klahnakoski commented on src/components/fileviewer.js:
```diff
@@ -33,29 +33,37 @@ export default class FileViewerContainer extends Component {
+          console.log(this.state.coverage.data[0][0].source.file.percentage_covered);
+        });
+      })
+      .catch(err => {
```
...also move this catch into the `getFileRevisionCoverage` function

klahnakoski commented on src/components/fileviewer.js:
```diff
@@ -33,29 +33,37 @@ export default class FileViewerContainer extends Component {
+          console.log('Error status code' + response.status);
+          return;
+        }
+        response.json().then(data => {
```
... also move the `.json()` call into the `getFileRevisionCoverage` function

klahnakoski commented:
Since the code should be moved to the `getFileRevisionCoverage` function, and this PR can not be merged.  I will leave it to you to update this before merging.

LinkaiQi commented:
I assume lines 22-34 also need to be moved to```getRawFile``` function as well?
```
    FetchAPI.getRawFile(revision, path)
      .then(response => {
        if (response.status !== 200) {
          console.log('Error status code' + response.status);
          return;
        }
        response.text()
      .then((text) => this.setState(() => ({ parsedFile: text.split("\n") })))
      .catch((error) => {
        console.error(error);
        this.setState(() => ({ appError: 'We did not manage to parse the file correctly.' }));
      });
    });
```

LinkaiQi committed:
Issue 16, simplify http call (refactoring)

Move http call funtion body(error handling, catch, json parser) to file_data.js
Bug fixing: fix bug caused by PR\#5 (extra slash in the 'path' variable)

LinkaiQi commented:
Hi Kyle Jen, 

The Code is updated
( contain bug fix cause by pervious PR #18 )

klahnakoski commented:
I will merge this now, so we minimize the merge comflicts
